azure/scope/managedcontrolplane.go

```diff
@@ -504,7 +492,7 @@ func (s *ManagedControlPlaneScope) ManagedClusterSpec() azure.ASOResourceSpecGet
 		DNSServiceIP:      s.ControlPlane.Spec.DNSServiceIP,
 		VnetSubnetID: azure.SubnetID(
 			s.ControlPlane.Spec.SubscriptionID,
-			s.VNetSpec().ResourceGroupName(),
+			s.Vnet().ResourceGroup,
 			s.ControlPlane.Spec.VirtualNetwork.Name,
 			s.ControlPlane.Spec.VirtualNetwork.Subnet.Name,
```

Comment:
Should these be also using `s.Vnet()`? Just to keep things consistent with the line above.

Reply:
`s.Vnet()` doesn't define the `Subnet`, and all the other places using the vnet resource group refer to it like this, so it seems like there's some inconsistency necessary here without some wider refactoring.